src/lib.rs

```diff
@@ -195,13 +198,37 @@ impl Labelset {
         K: AsRef<[u8]>,
         V: AsRef<[u8]>,
     {
-        unsafe {
+        let errno = unsafe {
             sys::labelset_set(
                 self.raw.as_ptr(),
                 key.as_ref().into(),
                 value.as_ref().into(),
             )
         };
+        if errno != 0 {
+            panic!("corruption in custom labels library: errno {errno}");
```

Comment:
What inspired this change? This happens because `malloc` failed, i.e. in OOM conditions, not because of internal corruption.

I agree it's good to catch this but the panic message should be changed. Actually I'm not convinced panicking is even the right thing to do here but I could be convinced.

Reply:
Just realized we weren't checking `errno` here. I changed the panic error message to "out of memory", which matches what the library already does in other situations where `malloc` returns `NULL`: https://github.com/polarsignals/custom-labels/blob/master/src/lib.rs#L88